### PR TITLE
second request with fixes and @CaseyCarter branch merged

### PR DESCRIPTION
**Pull request by JeffGarland**

### CODE REVIEW

**Comment by CaseyCarter** [comment]
For future reference: you can simply add commits to your branch, or even force push on top of it, and Github will pickup the changes for the PR. There's no need to open a new PR.

Let's keep our fingers crossed that the clang with modules build succeeds this time so I don't have to figure out why it's breaking.

**Comment by CaseyCarter** [comment]
@ericniebler, could you please kill https://travis-ci.org/ericniebler/range-v3/builds/300784955 from the old PR so this PR can start CI?

**Comment by ericniebler** [comment]
> could you please kill https://travis-ci.org/ericniebler/range-v3/builds/300784955 

Done.

**Comment by CaseyCarter** [comment]
> > could you please kill https://travis-ci.org/ericniebler/range-v3/builds/300784955
>
> Done.

FYI: If you were to create a Github organization, transfer the range-v3 repo into it, and invite others to the org, we could deal with transient CI problems like this. (I don't know if it's worth the trouble, but I thought I'd mention it.)

**Comment by JeffGarland** [comment]
@CaseyCarter thx for the tip, not that experience on the pull requests.  As soon as we work this out I have a larger batch to commit.

File: doc/index.md [hunk 1]
```diff
@@ -330,7 +330,7 @@ two overloads are guaranteed to not be ambiguous.
 
 Range v3 forms the basis for a proposal to add ranges to the standard library
 ([N4128](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4128.html)),
-and is also be the basis for a Technical Specification on Ranges. The Technical 
+and is also be the basis for a Technical Specification on Ranges. The Technical
```
Comment (ericniebler) [inline doc/index.md]: "and is also be" ? :-P You don't have to fix it, I will.

**Comment by CaseyCarter** [comment]
Help me, @EricWF , you're my only hope!

Do you have any idea why the clang + modules build is emitting a warning here despite that it is suppressed? The non-module builds are behaving properly. I hate to play the compiler bug card, but I have no other ideas and I don't know how to reproduce this locally.

(@ericniebler: I suggest we merge this to unblock @JeffGarland and simply leave the clang+modules build broken until we can investigate the issue.)

**Comment by ericniebler** [comment]
> (@ericniebler: I suggest we merge this to unblock @JeffGarland and simply leave the clang+modules build broken until we can investigate the issue.)

Can we exclude the examples from the clang modules build?

**Comment by CaseyCarter** [comment]
Yes, but I'm about to board my flight to SEA, so it may be a bit before I get to it. 

(Another idea is to include the crazy warning stuff in the test directory instead of at the top level, so it is never applied to the other top level directories.)

**Comment by CaseyCarter** [comment]
> Another idea is to include the crazy warning stuff in the test directory instead of at the top level, so it is never applied to the other top level directories.

This is the fix I went with (https://github.com/JeffGarland/range-v3/compare/master...CaseyCarter:split_warnings), it's in CI (https://travis-ci.org/CaseyCarter/range-v3/builds/301167552) and I will PR into this branch when it completes cleanly.

**Comment by EricWF** [comment]
@CaseyCarter Can you point me to exactly which error you're referring?

**Comment by JeffGarland** [comment]
I finally got around to pulling @CaseyCarter changes -- basically he spit the warning setup for the tests versus examples.

**Comment by CaseyCarter** [comment]
> @CaseyCarter Can you point me to exactly which error you're referring?

@EricWF Sorry for paging you without providing a direct link. This Travis build (https://travis-ci.org/ericniebler/range-v3/jobs/300786518) illustrates the error nicely:
<pre>
cd /home/travis/build/ericniebler/range-v3/build/example && /usr/bin/clang++-5.0   -I/home/travis/build/ericniebler/range-v3/include  -stdlib=libc++ -nostdinc++ -cxx-isystem /usr/include/c++/v1/ -Wno-unused-command-line-argument -fsanitize=memory -O3 -DNDEBUG   -std=c++1z -Werror -Wall -Wextra -Weverything -pedantic -pedantic-errors -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-weak-vtables -Wno-padded -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-documentation -Wno-documentation-unknown-command -Wno-old-style-cast -Wno-shadow -D__extern_always_inline=inline -fdiagnostics-show-template-tree -ftemplate-backtrace-limit=0 -fmodules -fmodule-map-file=/home/travis/build/ericniebler/range-v3/include/module.modulemap -fmodules-cache-path=/home/travis/build/ericniebler/range-v3/build/module.cache -fmodule-map-file=/usr/include/c++/v1/module.modulemap -fsanitize=memory -fsanitize-memory-track-origins -fno-sanitize-recover=all -fno-omit-frame-pointer -Ofast -fstrict-aliasing -fstrict-vtable-pointers -ffast-math -fvectorize -fsized-deallocation -msse4.2 -mtune=native -Wno-missing-braces -Wno-shorten-64-to-32 -Wno-global-constructors -o CMakeFiles/count.dir/count.cpp.o -c /home/travis/build/ericniebler/range-v3/example/count.cpp
/home/travis/build/ericniebler/range-v3/example/count.cpp:36:26: error: suggest
      braces around initialization of subobject [-Werror,-Wmissing-braces]
  std::array<int, 6> a { 6, 2, 3, 4, 5, 6 };
                         ^~~~~~~~~~~~~~~~
                         {               }
1 error generated.
</pre>
Despite that `-Wno-missing-braces` is right there near the end of the command line, the compiler diagnoses missing braces. (Be sure to use the commit linked from the Travis build if you want to repro, this branch has changed since and no longer reproes the bug.)

**Comment by ericniebler** [comment]
Great, thanks guys. I'll update the official docs when I get back on a proper net connection.

**Comment by JeffGarland** [comment]
Ok -- I have a second batch in work, so hopefully soon :)

**Comment by ericniebler** [comment]
The updated docs are now live at https://ericniebler.github.io/range-v3/.

**Pull request merged**